workflow-templates/check-license.yml

```
@@ -78,23 +94,28 @@ jobs:
       - name: Install licensee
         run: gem install licensee
 
-      - name: Check license file
+      - name: Check license file for ${{ matrix.check-license.path }}
         run: |
           EXIT_STATUS=0
+
+          # Go into folder path
+          cd ./${{ matrix.check-license.path }}
+
           # See: https://github.com/licensee/licensee
           LICENSEE_OUTPUT="$(licensee detect --json --confidence=100)"
 
+
           DETECTED_LICENSE_FILE="$(echo "$LICENSEE_OUTPUT" | jq .matched_files[0].filename | tr --delete '\r')"
           echo "Detected license file: $DETECTED_LICENSE_FILE"
-          if [ "$DETECTED_LICENSE_FILE" != "\"${EXPECTED_LICENSE_FILENAME}\"" ]; then
-            echo "::error file=${DETECTED_LICENSE_FILE}::detected license file $DETECTED_LICENSE_FILE doesn't match expected: $EXPECTED_LICENSE_FILENAME"
+          if [ "$DETECTED_LICENSE_FILE" != "\"${matrix.check-license.EXPECTED_LICENSE_FILENAME}\"" ]; then
+            echo "::error file=${DETECTED_LICENSE_FILE}::detected license file $DETECTED_LICENSE_FILE doesn't match expected: $matrix.check-license.EXPECTED_LICENSE_FILENAME"
```

```suggestion
          if [ "$DETECTED_LICENSE_FILE" != "\"${{ matrix.check-license.EXPECTED_LICENSE_FILENAME }}\"" ]; then
            echo "::error file=${DETECTED_LICENSE_FILE}::detected license file $DETECTED_LICENSE_FILE doesn't match expected: ${{ matrix.check-license.EXPECTED_LICENSE_FILENAME }}"
```

Use the correct syntax when referencing GitHub Actions contexts:

https://docs.github.com/en/actions/learn-github-actions/contexts#:~:text=You%20can%20access%20contexts%20using%20the%20expression%20syntax

It can be confusing because the syntax for referencing shell variables looks similar, but these are two completely different things. The GitHub Actions framework is a templating system, where the expressions are replaced before the resulting code is executed in the shell.